src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -69,16 +69,51 @@
 
   public static final String JDBC_TTL = "net.snowflake.jdbc.ttl";
 
-  /** The unique httpClient shared by all connections. This will benefit long- lived clients */
-  private static Map<OCSPMode, CloseableHttpClient> httpClient = new ConcurrentHashMap<>();
+  /**
+   * Object to provide the proxy user and password credentials for the HTTP Client object when
+   * needed
+   */
+  private static CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
 
   /**
-   * The unique httpClient shared by all connections that don't want decompression. This will
-   * benefit long-lived clients
+   * The unique httpClient shared by all connections. This will benefit long- lived clients. Key =
+   * proxy host + proxy port + nonProxyHosts, Value = Map<OCSPMode, HttpClient>
    */
-  private static Map<OCSPMode, CloseableHttpClient> httpClientWithoutDecompression =
+  private static Map<String, Map<OCSPMode, CloseableHttpClient>> httpClientWithProxies =
```

Comment:
This is now Map<String, Map<OCSPMode, CloseableHttpClient>> because it's sorted by proxy AND OCSP mode. Since there are only 3 OCSP modes, I don't think this will be a performance degradation.